stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A value that represents either a success or failure, capturing associated
+/// values in both cases.
+@_frozen
+public enum Result<Success, Failure: Error> {
+  /// A success, storing a `Success` value.
+  case success(Success)
+  
+  /// A failure, storing a `Failure` value.
+  case failure(Failure)
+  
+  /// Evaluates the given transform closure when this `Result` instance is
```

Comment:
I'm not sure what's gained by adding additional terminology to describe the closure as compared to what's used in documenting `Optional.map`.

```suggestion
  /// Evaluates the given closure when this `Result` instance is
```

Reply:
Updated.